src/results/resultTable.ts

```diff
@@ -0,0 +1,385 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { StructRowProxy, Table } from 'apache-arrow';
+import { Table as PrintTable } from 'console-table-printer';
+
+import { ArrowRelation } from '../api/transaction/types';
+import { convertValue, getDisplayValue, getTypeDef } from './resultUtils';
+import { ConstantValue, RelTypeDef, RelTypedValue } from './types';
+
+export interface ResultColumn {
+  /**
+   * Returns type definition of the column
+   *
+   * @returns Type definition.
+   */
+  typeDef: RelTypeDef;
+
+  /**
+   * Returns an iterator over column values.
+   *
+   * @returns An iterator over column values.
+   */
+  [Symbol.iterator](): IterableIterator<RelTypedValue['value']>;
+
+  /**
+   * Returns column values as an array
+   *
+   * @returns An array of column values.
+   */
+  values: () => RelTypedValue['value'][];
+
+  /**
+   * Return a value at the given index.
+   *
+   * @param {string} index Row index.
+   * @returns Value or undefined if the index is out of range.
+   */
+  get: (index: number) => RelTypedValue['value'] | undefined;
+
+  /**
+   * Number of values in the column.
+   *
+   * @returns Number of values.
+   */
+  readonly length: number;
+}
+
+interface IteratorOf<T> {
+  [Symbol.iterator](): IterableIterator<T>;
+  values: () => T[];
+  readonly length: number;
+}
+
+type ColumnDef = {
+  typeDef: RelTypeDef;
+  arrowIndex?: number;
+};
+
+/**
+ * ResultTable provides an interface over {@link ArrowRelation} that maps Rel
+ * types to their corresponding JavaScript equivalents.
+ */
+export class ResultTable implements IteratorOf<RelTypedValue['value'][]> {
+  private table: Table;
+  private colDefs: ColumnDef[];
+
+  /**
+   * Instantiate a new ResultTable instance.
+   *
+   * @example
+   *   cosnt result = await client.exec('database', 'engine', 'def output = 123, "test"')
+   *   cosnt table = new ResultTable(result.results[0]);
+   *
+   *   console.log(table.values()); // Prints [[123n, "test"]];
+   *
+   * @param relation Arrow relation
+   */
+  constructor(private relation: ArrowRelation) {
+    this.table = relation.table;
+
+    const types = relation.relationId.split('/').filter(x => x);
+
+    let arrowIndex = 0;
+
+    this.colDefs = types.map(t => {
+      const typeDef = getTypeDef(t);
+
+      const colDef: ColumnDef = { typeDef };
+
+      if (typeDef.type !== 'Constant') {
+        colDef.arrowIndex = arrowIndex;
+        arrowIndex++;
+      }
+
+      return colDef;
+    });
+  }
+
+  /**
+   * Array for type definitions per column. Shortcut for column.typeDef.
+   *
+   * @returns An array of type definitions.
+   */
+  get typeDefs() {
```

Comment:
Should this just be a function instead of a getter? Seems similar to `columns` to me.